### PR TITLE
Improve handling of FIG viewport resizing

### PR DESCRIPTION
**Pull request by contolini**
Currently, the interactive FIG desktop side nav and mobile accordion nav are initialized on page load based on the user's viewport width. They are not re-initialized if the user changes the width of their browser. This PR uses [`matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) to listen for viewport resizing and enable/disable FIG features as needed.

I also instructed the FIG browser tests to not run unless a `CYPRESS_FIG_URL` environment variable is defined because the FIG URL is a moving target until we launch so it probably shouldn't be hardcoded.

## Changes

- Moved the FIG mobile nav logic into its own function so it can be enabled/disabled if/when the user resizes their browser window.
- Simplified the FIG browser test file to skip the FIG tests if the user hasn't defined a `CYPRESS_FIG_URL` environment variable.

## Additions

- `CYPRESS_FIG_URL` environment variable that has to be defined or the FIG tests won't run (which is good because most of us don't have a FIG page on our localhost).

## How to test this PR

1. Units tests should pass: `yarn jest fig-sidenav-utils-spec.js`
1. Functional tests should pass headlessly:
    `yarn cypress run --spec test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.cy.js --env FIG_URL=/data-research/small-business-lending/filing-instruction-guide/filing-instructions-guide-test/ --config baseUrl=https://<secret-dev4-subdomain>.cfpb.gov`
1. Functional tests should pass with your local copy of Chrome:
    `yarn cypress open --e2e --browser chrome --env FIG_URL=/data-research/small-business-lending/filing-instruction-guide/filing-instructions-guide-test/ --config baseUrl=https://<secret-dev4-subdomain>.cfpb.gov `

## Notes and todos

-

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.cy.js [hunk 1]
```diff
@@ -1,223 +1,214 @@
...
-} );
+describe( '1071 Filing Instruction Guide (FIG)', () => {
+
+  describe( 'FIG table of contents', () => {
+
+    /* Our FIG sample page only exists in certain environments so continue
+       this test suite only if the user explicitly provides a URL via
+       CYPRESS_FIG_URL=https://blah.cfpb.gov/xxxxxx/yyyyyy/zzzzzz/.
+       Once the FIG reaches production we can disable this check. */
+    onlyOn( Boolean( fig.url() ), () => {
```
Comment (niqjohnson) [inline test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.cy.js:14]: I know this was done to variable-ize the FIG URL, but I love the added bonus of how much simpler the skip test logic has become by using an environment variable.

Comment (contolini) [reply test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.cy.js:14]: Thank you btw @niqjohnson for fixing these tests in the first place and preventing them from breaking our CI when the FIG isn't available.

**Review by schbetsy — COMMENTED** [review]
Tests pass and resizing looks great now! Is the search function now busted at mobile sizes, or is it just me? Edit: Oh, I see it's not busted, but the TOC pane stays covering the page after I click on a search result.

**Comment by contolini** [comment]
> Tests pass and resizing looks great now! Is the search function now busted at mobile sizes, or is it just me? Edit: Oh, I see it's not busted, but the TOC pane stays covering the page after I click on a search result.

Ah yes! There are no browser tests covering the new search feature. I'll create a separate ticket to capture that and this bug.

**Review by schbetsy — APPROVED** [review]

**Pull request merged**